### DISCUSSION TITLE
Transforming constants with lodash breaks the translations

### DISCUSSION THREAD

**Discussion by manuganji** [lingui/js-lingui#1118] [General]
```
// constants.js

export const OFFER_TYPES = {
  RENT: {
    code: "rent",
    label: t`Rent`,
  },
  SALE: {
    code: "sale",
    label: defineMessage({ id: `Sale`, message: t`Sale` }), // t`Sale`,
  },
};

// This works.
export const codeToOfferTypeLabel = Object.fromEntries(
   Object.values(OFFER_TYPES).map((type) => [type.code, type.label])
);


// This doesn't work.
export const codeToOfferTypeLabel = _.fromPairs(
  _.values(OFFER_TYPES).map((type) => [type.code, type.label])
);
```

I later access `{i18n._(codeToOfferTypeLabel[listing.offerType])}` within React. Only one of the variants works. I believe this has something to do with lodash's internal DS or with the way modules are exported.

**Comment by manuganji** [comment]
They both work in `DEV` but only the `Object` variant works to produce build. I'm using "next.js"

**Reply by semoal** [reply]
To use translations on constants files you must use **defineMessage**, feel free to check documentation.
I'm off for holidays, I'll take Lingui control in a few days.

**Reply by manuganji** [reply]
Thank you so much for replying on a holiday, @semoal. My build is again not working.  I'll try your suggestion out.
